staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go

```diff
@@ -88,6 +88,47 @@ func TestInt64AmountAdd(t *testing.T) {
 		}
 	}
 }
+
+func TestInt64AmountMul(t *testing.T) {
+	for _, test := range []struct {
+		a  int64Amount
+		b  int64
+		c  int64Amount
+		ok bool
+	}{
+		{int64Amount{value: 100, scale: 1}, 1000, int64Amount{value: 100000, scale: 1}, true},
```

Comment:
I feel like I'm a pest, but these cases still feel sort of haphazard, can we be a little more systematic?  E.g. something like:

https://go.dev/play/p/k4IOkuEvrXD

That may be a bit much, but I hope you see what I mean - I'm staring at these big tables of struct initisalisations, some with function calls in-between and trying to convince myself that all the edge case are covered.  Like, 0 * 0 isn't covered.  Probably should be.

Reply:
I really appreciate your review :)
I'll try to make tests more systematic and add a test case for `0 * 0`.

Reply:
@thockin I tried to make test cases more systematic. However, because we need to define expected `ok` against test cases in advance, it was hard to make test cases more systematic using the way you suggested.

So, how about just adding a formula as a test name like this:

```go
"100×10^1 × 1000 = 100000×10^1": {int64Amount{value: 100, scale: 1}, 1000, int64Amount{value: 100000, scale: 1}, true}
"1×10^100 × 10 = OVERFLOW": {int64Amount{value: 1, scale: 100}, 10, int64Amount{value: 1, scale: 100}, false},
```

Although this way couldn't make test cases systematic, I believe that this way makes it easier to find leaking edge cases.

Reply:
I love the idea of adding useful names

Reply:
Maybe also, for error cases, make t.c be `int64Amount{}` since we the test loop should not even look at it.

Reply:
> Maybe also, for error cases, make t.c be int64Amount{} since we the test loop should not even look at it.

SGTM